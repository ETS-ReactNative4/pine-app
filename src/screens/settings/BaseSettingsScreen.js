import React, { Component } from 'react';
import { StyleSheet, ScrollView, View } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ErrorBoundary from '../../components/ErrorBoundary';

const styles = StyleSheet.create({
  view: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: '#EFEFF4'
  },
  childrenWrapper: {
    paddingTop: 30
  }
});

export default class BaseSettingsScreen extends Component {
  render() {
    return (
      <ErrorBoundary {...this.props} style={styles.view}>
        <ScrollView>
          <View style={styles.childrenWrapper}>
            {this.props.children}
          </View>
        </ScrollView>
      </ErrorBoundary>
    );
  }
}

BaseSettingsScreen.propTypes = {
  children: PropTypes.node
};
