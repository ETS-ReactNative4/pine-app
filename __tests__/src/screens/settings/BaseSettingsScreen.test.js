import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BaseSettingsScreen from '../../../../src/screens/settings/BaseSettingsScreen';

describe('BaseSettingsScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BaseSettingsScreen />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
