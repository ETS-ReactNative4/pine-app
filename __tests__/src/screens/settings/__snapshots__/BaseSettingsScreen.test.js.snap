// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSettingsScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#EFEFF3",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingTop": 35,
          }
        }
      />
    </View>
  </RCTScrollView>
  <ErrorModalContainer />
</View>
`;
